### PR TITLE
Reader: Fix up how the Infinite List resets across queries.

### PR DESCRIPTION
**Pull request by blowery**
Need to clean out the sizing cache.

### CODE REVIEW

**Comment by blowery** [comment]
cc @TooTallNate

**Comment by blowery** [comment]
Fixes moving from query to query on reader search. To test, hit https://wpcalypso.wordpress.com/read/search?q=steph%20curry then change the query in page to something like `pants` and then to `party`. After each set of results loads, notice you can't really scroll down past 4 items.

Now try it on this branch and notice it works as expected.

**Comment by blowery** [comment]
cc @aduth too because

File: client/reader/following-stream/index.jsx [hunk 1]
```diff
@@ -169,6 +169,7 @@ module.exports = React.createClass( {
 			this.props.store.off( 'change', this.updateState );
 			nextProps.store.on( 'change', this.updateState );
 			this.updateState( nextProps.store );
+			this._list.reset();
```
Comment (blowery) [inline client/reader/following-stream/index.jsx]: should probably guard this.

File: client/components/infinite-list/index.jsx [hunk 2]
```diff
@@ -161,6 +161,27 @@ module.exports = React.createClass( {
 		}
 	},
 
+	reset: function() {
```
Comment (blowery) [inline client/components/infinite-list/index.jsx]: i do not all this duplication with componentWillMount, but initial attempts to DRY it out made it worse. The state depending on a call to the scrollHelper which is being created here complicates it.

Part of me wants to ditch scrollHelper and yank it back into index. Breaking it out just makes life more complicated.

Comment (rralian) [inline client/components/infinite-list/index.jsx]: I would love to rip out a lot of infinite-scroll logic. I think we merged it (or rather I merged it) without really quantifying the benefit.  But the increased complexity is daunting.

Comment (blowery) [inline client/components/infinite-list/index.jsx]: i ain't doing it here :)

**Comment by blowery** [comment]
reviewed by @TooTallNate

**Pull request merged**